src/graph/test/DeleteEdgeTest.cpp

```diff
@@ -0,0 +1,223 @@
+        }
+        meta::TestUtils::registerHB(network::NetworkUtils::toHosts(host).value());
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE SPACE mySpace(partition_num=1, replica_factor=1)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    sleep(FLAGS_load_data_interval_secs + 1);
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "USE mySpace";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE TAG person(name string, age int)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE EDGE friend(intimacy int)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "CREATE EDGE schoolmate(likeness int)";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    sleep(FLAGS_load_data_interval_secs + 1);
+    return TestOK();
+}
+
+AssertionResult DeleteEdgeTest::removeData() {
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "DROP SPACE mySpace";
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = folly::stringPrintf("REMOVE HOSTS 127.0.0.1:%u",
+                                              gEnv->storageServerPort());
+        auto code = client_->execute(cmd, resp);
+        if (cpp2::ErrorCode::SUCCEEDED != code) {
+            return TestError() << "Do cmd:" << cmd << " failed";
+        }
+    }
+
+     return TestOK();
+}
+
+TEST_F(DeleteEdgeTest, DeleteEdges) {
+    // Insert vertices and edges
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age) VALUES 1001:(\"Zhangsan\", 22),"
+                          "1002:(\"Lisi\", 23), 1003:(\"Jack\", 18), 1004:(\"Rose\", 19)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE friend(intimacy) VALUES 1001->1002@15:(90),"
```

Comment:
add two (different types or different ranks) edgess with the same src and the same dst. Then delete one . 

and try to double delete the same edge.

delete a not exist edge. 

check: both the in and out keys are not exists in storage ( not partially deleted)

Reply:
Ok